sdks/java/core/src/test/java/org/apache/beam/sdk/util/ApiSurfaceVerificationTest.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util;
+
+import static org.hamcrest.Matchers.anyOf;
+import static org.junit.Assert.fail;
+
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Ordering;
+import com.google.common.collect.Sets;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This test aims at keeping the public API is conformant to a hard-coded policy by
+ * testing whether a package (determined by the location of the test) exposes only white listed
+ * packages/classes.
+ * <p>
+ * Tests that derive from {@link ApiSurfaceVerificationTest} should be placed under the package to
+ * be * tested and implement {@link ApiSurfaceVerificationTest#allowedClasses()}.
+ * Further customization can be done by overriding {@link ApiSurfaceVerificationTest#apiSurface()}.
+ * </p>
+ */
+// based on previous code by @kennknowles and others.
+@RunWith(JUnit4.class)
+public abstract class ApiSurfaceVerificationTest {
```

Comment:
With the inheritance removed, this doesn't need to be a JUnit test suite at all, but just a place for utilities. It could reasonably be merged with `ApiSurface` itself.